modules/akka/src/main/scala/com/github/eikek/calev/akka/CalevActorScheduling.scala

```diff
@@ -0,0 +1,37 @@
+package com.github.eikek.calev.akka
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.scaladsl.ActorContext
+import com.github.eikek.calev.CalEvent
+import com.github.eikek.calev.internal.{DefaultTrigger, Trigger}
```

Comment:
These internal classes are not recommended to use. I don't know akka typed - so I can't judge if this is now a good example to use it. If it is possible to only rely on `CalEvent.nextElapse` I'd prefer this.

Reply:
I did not expect CalEvent#nextElapse method to exists. Frankly speaking I see no reason to couple the the Trigger logic with CalEvent class. I would suggest to keep things simple and separate the "scheduling logic" from the CalEvent class. CalEvent should IMO be a simple case class. Trigger should be exposed to outside word (perhaps the name of the interface could be changed. I'm not sure but perhaps CalevScheduler name would be better. 

If backward compatibility is important than of course the existing methods in CalEvent should stay (perhaps marked as deprecated).

Please take my suggestions into consideration.

Reply:
Actually, CalEvent#nextElapse could still be supported but it should be implemented outside of CalEvent class (using scala DSL / implicits)

Reply:
Hi @pawelkaczor, thank for your input. To also be frank, the method is mentioned in the docs right in the README ;-). I see your point that calculating the next matching date could be outside the `CalEvent` type, but I don't feel it adds a lot of value here. The logic is not implemented directly in the `CalEvent` class, anyways. It delegates to the internal implementation in `DefaultTrigger`. I'd still consider it a "simple case class" although it has a few methods defined. In my opinion it is not "simpler" to introduce another interface to the user. And I don't think it's a better experience to have to add another import to get this feature via an implicit extension method. After all, this is a tiny library, where the only features are parsing calendar events and generating the next matching date. It's not at all a scheduler, nothing is scheduled - only a  date-time value is returned; i.e. there is no "scheduling logic" at all. As another example, the "parsing logic" is also accessible from within the companion object, this should then be moved out, too, and a `Parser` interface could be exposed. But here I also don't feel that it makes the library simpler to use. I don't expect multiple different parser implementations to be needed. You usually don't "work" with `CalEvent` instances, most of the time one is only interested in the next matching date, I would think.

So in my opinion it's fine to have this method directly on the `CalEvent` class - it's the only reason this class and library exists, so I like it to have this main feature on the main data type. (There is a DSL for creating calendar events in code that can be pulled in via implicits, but this is an optional feature.)

Reply:
Let me try again. I'm perfectly fine with the current library API / DSL. 
```
import com.github.eikek.calev._

val myCalEvent = CalcEvent.unsafe("....")

val nextEventInstant = myCalEvent.nextElapse(Instant.now)

```

The code above is perfectly fine.
The problem is that currently CalEvent case class implementation directly depends on com.github.eikek.calev.internal.DefaultTrigger. 

I would propose to remove this dependency. CalEvent should be simple, shareable class (that could easily be moved to a separate module, if needed, that does not depend on DefaultTrigger).

Reply:
I think I still do not to understand the big advantage. Why is CalEvent not shareable (what do you mean by this)? The dependency doesn't leak  (I hope). I think it is reasonable simple (there are 4 fields, 8 methods, alltogether 66loc). The Trigger/DefaultTrigger is an *implementation detail*, that shouldn't bother users actually (the name `internal` is intentional). Why is this a problem? I don't plan to move the code that calculates the next date into a separate module. If it is removed, then it is necessary to expose some interface which would be in the same core module. 
In cases like JSON I can see the advantage, because you might work with JSON values in general, and there *are* different parser implementations. But here, I can't see an advantage. Do you think there should be multiple ways to calculate the next matching date (and therefore multiple modules)? The CalEvent is only there to be able to get the next date, I don't think it is of much use outside of this, no? Or maybe I don't understand the problem that is being solved this way?

Reply:
>> Do you think there should be multiple ways to calculate the next matching date (and therefore multiple modules)?

I don't know, but for some reason you created trait Trigger that looks like service interface and DefaultTrigger that is the default implementation of the Trigger interface. And this is good, except that Trigger should not be kept in "internal" package.

Reply:
>> Why is this a problem?
I will try to refactor the PR (on a separate branch) and not use Trigger interface. If I find no problems than I won't insist on any changes on your side.

Reply:
> Trigger that looks like service interface and DefaultTrigger that is the default implementation of the Trigger interface

Ah ok, oh I think this is a misunderstanding :/ It was never my intention to make it look like a service interface, sorry. I think I better understand what you mean now. Sure, if it was a service interface meant to be usable, this would be the wrong place and then it makes much more sense. But it really never was thought that way. I guess I wasn't quite sure how many additional functions etc. I would be using. I sometimes start with a trait to see what I really want and then go playing around in an "implementation" - it is more meant as an internal thing. I think I can just remove this trait actually!



> I will try to refactor the PR (on a separate branch) and not use Trigger interface. If I find no problems than I'm fine with the current implementation on your side.

I would really appreciate this, thank you very much! I'm sorry for this confusion…. If you find problems, I definitely like to hear them, to better understand how they arrived with the current api.

Reply:
I will think of CalEvent more as a DSL than as an ordinary case class. 
As long as it is used only as a DSL and not as a data container that can be stored somewhere (in json for example) than I agree that it does not matter how it is implemented.